File: rust/src/bin/3612_process_string_with_special_operations_I.rs
impl Solution {
    pub fn process_str(s: String) -> String {
        let mut result = String::new();

        for c in s.chars() {
            match c {
                '*' => {
                    result.pop();
                }
                '#' => {
                    result += &result.clone();
                }
                '%' => {
                    result = result.chars().rev().collect::<String>();
                }
                _ => {
                    result.push(c);
                }
            }
        }

        result
    }
}

struct Solution {}

File: java/src/solutions/3612_process_string_with_special_operations_I.java
class Solution {
  public String processStr(String s) {
    StringBuilder result = new StringBuilder();

    for (int c : s.chars().toArray()) {
      switch (c) {
        case int x when x == '*' && result.length() > 0 -> result.deleteCharAt(result.length() - 1);
        case '*' -> {}
        case '#' -> result.append(result.toString());
        case '%' -> result.reverse();
        default -> result.append((char) c);
      }
    }

    return result.toString();
  }
}
